code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -87,14 +87,20 @@
 					L.Weaken(3) //you can't get away now~
 
 		if(grasping.len < max_grasps)
-			for(var/mob/living/L in range(grasp_range,src))
-				if(L == src || faction_check(L))
-					continue
-				if(!(L in grasping) && L != target && prob(grasp_chance))
-					L << "<span class='userdanger'>\the [src] has you entangled!</span>"
-					grasping[L] = Beam(L,"vine",'icons/effects/spacevines.dmi',INFINITY, 5,/obj/effect/ebeam/vine)
-
-					break //only take 1 new victim per cycle
+			grasping:
+				for(var/mob/living/L in view(grasp_range, src))
+					if(L == src || faction_check(L) || (L in grasping) || L == target)
```

Comment:
ya, byond has loop labels
```grasping:``` with the loop indented under it is the "loop label" of that loop, that loop is now 'called' grasping, which allows me to then use loop commands (```break```, ```continue```) on the specific named loop "grasping"